lib/svg/reaction_composer.rb

```diff
@@ -26,11 +26,25 @@ def initialize(materials_svg_paths, options = {})
     end
 
     def solvents_lines
-      s_per_line = @arrow_width < 150 ? 2 : 3
-      group_of_three = @solvents.each_slice(s_per_line).to_a
-      group_of_three.map do |i|
-        i.map{ |j| j && "#{j[0..7]}.." }
-      end.map{ |k| k.join(" / ") } unless @solvents.empty?
+      groups = @solvents.each_slice(num_solvents_per_line).to_a
+      groups.map do |g|
+        solvents_string(g)
+      end.map{ |k| k.join("  /  ") } unless @solvents.empty?
+    end
+
+    def num_solvents_per_line
+      reactants_size = @reactants.size > 0 ? @reactants.size : 1
+      reactants_size > 3 ? 3 : reactants_size
+    end
+
+    def solvents_string(group)
+      group.map do |j|
+        if j && j.size > 15
```

Comment:
When the string is longer than 15 characters, it will be limited to 15 characters with ```..```.
This is set by Nicole.